test/functional/owners_controller_test.rb

```diff
@@ -0,0 +1,213 @@
+require "test_helper"
+
+class OwnersControllerTest < ActionController::TestCase
+  context "When logged in" do
+    setup do
+      user = create(:user)
+      @rubygem = create(:rubygem)
+      create(:ownership, user: user, rubygem: @rubygem)
+      sign_in_as(user)
+    end
+
+    context "on GET to index" do
+      context "when user owns the gem" do
+        setup do
+          get :index, params: { rubygem_id: @rubygem.name }
+        end
+
+        should respond_with :success
+        should "render all gem owners in owners table" do
+          @rubygem.ownerships.each do |o|
+            assert page.has_content?(o.owner_name)
+          end
+        end
+      end
+
+      context "when user does not own the gem" do
+        setup do
+          @other_user = create(:user)
+          sign_in_as(@other_user)
+          get :index, params: { rubygem_id: @rubygem.name }
+        end
+
+        should redirect_to("the sign in page") { sign_in_path }
+      end
+    end
+
+    context "on POST to create ownership" do
+      context "with correct params" do
+        setup do
+          @new_owner = create(:user)
+          post :create, params: { handle: @new_owner.display_id, rubygem_id: @rubygem.name }
+        end
+
+        should redirect_to("ownerships index") { rubygem_owners_path(@rubygem) }
+        should "add unconfirmed ownership record" do
+          assert @rubygem.owners_including_unconfirmed.include?(@new_owner)
+          assert_nil @rubygem.ownerships_including_unconfirmed.find_by(user: @new_owner).confirmed_at
+        end
+        should "set success notice flash" do
+          expected_notice = "Owner added successfully. A confirmation mail has been sent to #{@new_owner.handle}'s email"
+          assert_equal expected_notice, flash[:notice]
+        end
```

Comment:
please add a test for mail being sent. Check EmailHelpers.

Reply:
test/integration/owner_test.rb:29
I have already added tests for mails. It more makes sense in integration test I feel.
wdyt?